docs/architecture/adr-015-namespace-id-size.md

```diff
@@ -0,0 +1,64 @@
+# ADR 15: Namespace ID Size
+
+## Status
+
+Proposed
+
+## Changelog
+
+- 2023/2/17: initial draft
+
+## Context
+
+Namespace ID is currently an 8 byte slice. 8 bytes provides a maximum of 2^64 possible namespace IDs. In practice some namespace IDs are reserved for protocol use so the number of namespace IDs available to users is 2^56 - 2. Modifying the size of a namespace ID post-launch is a breaking change and has implications for the NMT, share encoding, etc. so we'd like to carefully consider the size of the namespace ID pre-launch.
+
+Desirandum:
+
+1. Ability to randomly generate a new namespace that hasn't been used before
+
+// what other desirandum do we have?
+
+## Notes
+
+- The namespace ID must provide at least 72 bits of randomness ([Eager](https://eager.io/blog/how-long-does-an-id-need-to-be/)) to satisfy desirandum 1. Since an 8 byte namespace ID can only provide 64 bits of randomness, it fail to meet this desirandum.
+- [SHA256](https://en.wikipedia.org/wiki/SHA-2) has a digest size of 32 bytes so using a namespace ID size of 32 bytes would enable users to generate stable namespace IDs (e.g. `sha256('sov-labs')`) or unique namespace IDs (e.g. `sha256(blob)`) with the SHA256 hash function.
```

Comment:
> unique namespace IDs (e.g. `sha256(blob)`) with the SHA256 hash function.

It would be helpful to clarify that even in the case of sha256(blob), the namespace ID uniqueness still depends on the blob uniqueness. For example, if two blobs are identical, their hashes will also be identical.

Reply:
Resolved by https://github.com/celestiaorg/celestia-app/pull/1405/commits/1241dd04e85cf6097ed7d5c167cb5e7d8d9f2895